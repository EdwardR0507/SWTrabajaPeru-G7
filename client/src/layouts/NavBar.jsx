import React from "react";
import { withStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Link from "@material-ui/core/Link";
import Image from "material-ui-image";
import SearchField from "../components/TextFields/SearchField";
import theme from "../themes/themes";

const StyledNavBar = withStyles({
  root: {
    position: "relative",
    background: theme.palette.primary.main,
    height: "14.5%",
    lineHeight: "24px",
    letterSpacing: "0.18px",
    textTransform: "uppercase",
  },
})(AppBar);

const StyledTypography = withStyles({
  root: {
    color: "#FFFFFF",
    textDecoration: "none",
    margin: "0 20px",
  },
})(Typography);

const Logo = withStyles({
  root: {
    width: "96px",
    margin: "0 20px",
  },
})(Typography);
/*Cambiar por Image cuando esté listo el logo */

const NavBar = () => {
  return (
    <StyledNavBar>
      <Toolbar>
        <Logo />
        {/*Espacio vacío para el logo*/}
        <Link href="#">
          <StyledTypography variant="h5">Inicio</StyledTypography>
        </Link>
        <SearchField placeholder="Buscar Cliente o Servicio" />
        <Link href="#">
          <StyledTypography variant="h5">Registrarse</StyledTypography>
        </Link>
        <Link href="#">
          <StyledTypography variant="h5">Iniciar Sesión</StyledTypography>
        </Link>
      </Toolbar>
    </StyledNavBar>
  );
};

export default NavBar;
