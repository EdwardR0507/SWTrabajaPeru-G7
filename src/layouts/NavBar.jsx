import React, { useState } from "react";
import { useHistory } from "react-router";
import logo from "../assets/Trabaja peru.png";
import { withStyles } from "@material-ui/core/styles";
import { Link as RouterLink } from "react-router-dom";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Link from "@material-ui/core/Link";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import SearchField from "../components/TextFields/SearchField";
import IconButton from "@material-ui/core/IconButton";
import AccountCircleIcon from "@material-ui/icons/AccountCircle";
import EditIcon from "@material-ui/icons/Edit";
import ExitToAppIcon from "@material-ui/icons/ExitToApp";
import Avatar from "@material-ui/core/Avatar";
import theme from "../themes/themes";
import SideBar from "../layouts/Sidebar/SideBar";
import NotificationList from "../components/Lists/NotificationList";
import image from "../assets/worker.jpg";
const StyledNavBar = withStyles({
  root: {
    background: theme.palette.primary.main,
    display: "flex",
    position: "sticky",
    justifyContent: "space-around",
    height: "14.5%",
    lineHeight: "24px",
    letterSpacing: "0.18px",
    textTransform: "uppercase",
  },
})(AppBar);

const StyledToolbar = withStyles({
  root: {
    display: "flex",
    justifyContent: "space-between",
  },
})(Toolbar);

const StyledMenu = withStyles({})((props) => (
  <Menu
    elevation={0}
    getContentAnchorEl={null}
    anchorOrigin={{
      vertical: "bottom",
      horizontal: "center",
    }}
    transformOrigin={{
      vertical: "top",
      horizontal: "center",
    }}
    {...props}
  />
));

const StyledTypography = withStyles({
  root: {
    display: "inline",
    color: theme.colorLetter.primary.main,
    textDecoration: "none",
    marginRight: "40px",
    marginLeft: "25px",
  },
})(Typography);

const StyledIconButton = withStyles({
  root: {
    color: theme.colorLetter.primary.main,
    marginRight: "50px",
  },
})(IconButton);

const NavBar = (props) => {
  const [anchorEl, setAnchorEl] = useState(null);

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const history = useHistory();

  if (locaStorage.hasOwnProperty("User_session")){
    let token = localStorage.getItem("User_session")
    token = token.slice(1, -1)
  }

  return (
    <StyledNavBar>
      <StyledToolbar>
        {props.user ? (
          <>
            <>
              <SideBar />
              <img src={logo} alt="logo" width="56px" height="50px" />
            </>
            <SearchField
              placeholder="Buscar Cliente o Servicio"
              token={token}
            />
            <>
              <StyledIconButton>
                <NotificationList token={token} history={history} />
              </StyledIconButton>
              <IconButton onClick={handleClick}>
                <Avatar src={image} />
              </IconButton>
              <StyledMenu
                id="customized-menu"
                role="close"
                anchorEl={anchorEl}
                keepMounted
                disableScrollLock
                open={Boolean(anchorEl)}
                onClose={handleClose}
              >
                <MenuItem
                  onClick={() => {
                    history.push({
                      pathname: "/myAccount",
                      state: { token: token },
                    });
                  }}
                >
                  <ListItemIcon>
                    <AccountCircleIcon />
                  </ListItemIcon>
                  <ListItemText primary="Mi Perfil" />
                </MenuItem>
                <MenuItem
                  onClick={() => {
                    history.push({
                      pathname: "/editProfile",
                      state: { token: token },
                    });
                  }}
                >
                  <ListItemIcon>
                    <EditIcon />
                  </ListItemIcon>
                  <ListItemText primary="Editar Perfil" />
                </MenuItem>
                <MenuItem
                  onClick={() => {
                    localStorage.removeItem("User_session")
                    if (window.location.pathname === "/"){
                      window.location.reload();
                    }
                    else{
                      history.push({
                        pathname: "/",
                      });
                    }
                  }}
                >
                  <ListItemIcon>
                    <ExitToAppIcon />
                  </ListItemIcon>
                  <ListItemText primary="Cerrar Sesión" />
                </MenuItem>
              </StyledMenu>
              <Link>
                <StyledTypography variant="h5">
                  {props.user.us_nombres}
                </StyledTypography>
              </Link>
            </>
          </>
        ) : (
          <>
            <img src={logo} alt="logo" width="60px" height="50px" />
            <div>
              <Link component={RouterLink} to="/signup">
                <StyledTypography variant="h5">Registrarse</StyledTypography>
              </Link>
              <Link component={RouterLink} to="/signin">
                <StyledTypography variant="h5">Iniciar Sesión</StyledTypography>
              </Link>
            </div>
          </>
        )}
      </StyledToolbar>
    </StyledNavBar>
  );
};

export default NavBar;
