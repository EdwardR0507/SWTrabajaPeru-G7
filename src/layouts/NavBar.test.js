/*Importamos las librerias principales*/
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { shallow } from "enzyme";
import NavBar from "./NavBar";
/*Renderizado del componente SideBar*/
describe("NavBar", () => {
  it('render "NavBar"', () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
