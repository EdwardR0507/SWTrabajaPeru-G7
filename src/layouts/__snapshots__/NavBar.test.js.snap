// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar render "NavBar" 1`] = `ShallowWrapper {}`;
